synapseclient/models/agent.py

```diff
@@ -0,0 +1,629 @@
+                    instance from the Synapse class constructor.
+
+        Returns:
+            The retrieved AgentSession object.
+        """
+        session_response = await get_session(
+            id=self.id,
+            synapse_client=synapse_client,
+        )
+        return self.fill_from_dict(synapse_agent_session=session_response)
+
+    @otel_trace_method(
+        method_to_trace_name=lambda self, **kwargs: f"Update_Session: {self.id}"
+    )
+    async def update_async(
+        self,
+        *,
+        synapse_client: Optional[Synapse] = None,
+    ) -> "AgentSession":
+        """Updates an agent session.
+        Only updates to the access level are currently supported.
+
+        Arguments:
+            synapse_client: If not passed in and caching was not disabled by
+                    `Synapse.allow_client_caching(False)` this will use the last created
+                    instance from the Synapse class constructor.
+
+        Returns:
+            The updated AgentSession object.
+        """
+        session_response = await update_session(
+            id=self.id,
+            access_level=self.access_level,
+            synapse_client=synapse_client,
+        )
+        return self.fill_from_dict(synapse_agent_session=session_response)
+
+    @otel_trace_method(method_to_trace_name=lambda self, **kwargs: f"Prompt: {self.id}")
+    async def prompt_async(
+        self,
+        prompt: str,
+        enable_trace: bool = False,
+        print_response: bool = False,
+        newer_than: Optional[int] = None,
+        *,
+        synapse_client: Optional[Synapse] = None,
+    ) -> None:
+        """Sends a prompt to the agent and adds the response to the AgentSession's
+        chat history. A session must be started before sending a prompt.
+
+        Arguments:
+            prompt: The prompt to send to the agent.
+            enable_trace: Whether to enable trace for the prompt.
+            print_response: Whether to print the response to the console.
+            newer_than: The timestamp to get trace results newer than.
+                Defaults to None (all results).
+            synapse_client: If not passed in and caching was not disabled by
+                    `Synapse.allow_client_caching(False)` this will use the last created
+                    instance from the Synapse class constructor.
+        """
+
+        agent_prompt = await AgentPrompt(
+            prompt=prompt, session_id=self.id, enable_trace=enable_trace
+        ).send_job_and_wait_async(
+            synapse_client=synapse_client, post_exchange_args={"newer_than": newer_than}
+        )
+
+        self.chat_history.append(agent_prompt)
+
+        if print_response:
+            print(f"PROMPT:\n{prompt}\n")
+            print(f"RESPONSE:\n{agent_prompt.response}\n")
+            if enable_trace:
+                print(f"TRACE:\n{agent_prompt.trace}")
```

Comment:
Should these be raw `print` statements, instead of a `syn.logger.info()` message?